databuilder/extractor/bigquery_usage_extractor.py

```diff
@@ -0,0 +1,103 @@
+from collections import namedtuple
+from datetime import date, timedelta
+import logging
+
+from googleapiclient.discovery import build
+from oauth2client.client import GoogleCredentials
+
+from databuilder.extractor.base_extractor import Extractor
+
+TableColumnUsageTuple = namedtuple('TableColumnUsageTuple', ['database', 'cluster', 'schema',
+                                                             'table', 'column', 'email'])
+
+LOGGER = logging.getLogger(__name__)
+
+
+class BigQueryUsageExtractor(Extractor):
+    """
+    An aggregate extractor for bigquery table usage. This class takes the data from
+    the stackdriver logging API by filtering on timestamp, bigquery_resource and looking
+    for referencedTables in the response.
+    """
+    TIMESTAMP_KEY = 'timestamp'
+    PROJECT_ID_KEY = 'projectid'
+
+    def init(self, conf):
+        credentials = GoogleCredentials.get_application_default()
+        self.logging_service = build('logging', 'v2', credentials=credentials)
+
+        # type: (ConfigTree) -> None
+        self.timestamp = conf.get_string(
+            BigQueryUsageExtractor.TIMESTAMP_KEY,
+            (date.today() - timedelta(days=1)).strftime('%Y-%m-%dT00:00:00Z'))
+        self.projectid = conf.get_string(BigQueryUsageExtractor.PROJECT_ID_KEY)
+        self.table_usage_counts = {}
+        self._count_usage()
+        self.iter = iter(self.table_usage_counts)
+
+    def _count_usage(self):
+        for entry in self._retrieve_records():
```

Comment:
How about adding progress log if this takes long time?

Reply:
I added a logger for every "pagesize" records processed